substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -80,12 +80,10 @@ private StubForAbstractClass() {
 
     private static final class SerializationLookupKey {
         private final Class<?> declaringClass;
-        private final Class<?> targetConstructorClass;
 
-        private SerializationLookupKey(Class<?> declaringClass, Class<?> targetConstructorClass) {
-            assert declaringClass != null && targetConstructorClass != null;
+        private SerializationLookupKey(Class<?> declaringClass) {
```

Comment:
As already mentioned in our meeting today. This **breaks the serialization feature** and will not pass our internal CI gates. The correct way to fix your serialization issues for the benchmarks is to provide custom configs for special cases of serialization used by these benchmarks. The format for the serialization config entries can be found in https://github.com/oracle/graal/blob/f07d5e3b858b2441e942958d8f329238ff0b0c87/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/doc-files/SerializationConfigurationFilesHelp.txt#L14

Reply:
For such cases the serialization error message that you get at runtime contains all info that you need to create that custom bits of serialization config. https://github.com/oracle/graal/blob/89e4cfc7aeea69970b60c64cd075ceb2a104e864/substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java#L132

`declaringClass` provides the value for json key-`name` and customTargetConstructorClass provides the value for json key-`targetConstructorClass`

Reply:
Undrestand. I'll try to do it in another way.